pulpcore/tests/unit/serializers/test_repository.py

```diff
@@ -144,3 +145,23 @@ def test_validate_checkpoint_and_repository():
     data["publication"] = mock_publication
     with pytest.raises(serializers.ValidationError):
         serializer.validate(data)
+
+
+def test_validate_repo_remote_sync(monkeypatch):
+    """
+    Test that the synchronization of repository and remote fails if they are from different plugins.
+    """
+    mock_remote = Mock()
```

Comment:
Since `get_prn` does an `isinstance(instance, Model)` check you are going to need to specify the object the Mock represents. Do the same for the `mock_repo`
```suggestion
    mock_remote = Mock(spec=models.Remote)
```